kola/harness.go

```diff
@@ -109,8 +109,8 @@ func filterTests(tests map[string]*register.Test, pattern, platform string, vers
 			continue
 		}
 
-		// Check the test's minVersion when running more then one test
-		if version.LessThan(t.MinVersion) && t.Name != pattern {
+		// Check the test's min and max versions when running more then one test
+		if t.Name != pattern && t.MinVersion.LessThan(t.MaxVersion) && (!t.MinVersion.LessThan(version) || !version.LessThan(t.MaxVersion)) {
```

Comment:
Sorry, its unintuitive: `/kola/register/register.go`

That package exists to avoid a cyclic dependency